soccer/gameplay/skills/benchmark/motion_test.py

```diff
@@ -0,0 +1,100 @@
+from abc import ABC, abstractmethod
+import time
+
+
+#An abstract base class for motion tests to be run by subclasses of the BaseMotionBenchmark skill
+class MotionTest(ABC): 
+
+    def __init__(self, nRuns, benchmark):
+        self.runs = nRuns
+        self.theMotionBenchmark = benchmark
+        self.title = "No Name Test"
+
+        self.motionPause = 0.5
+
+        self.sParams = dict() #Scoring parameters
+        self.sResults = dict() #Raw results
+        self.sScore = dict() #Actual scores
+
+        #Is used to determine if the test has actually
+        #started, to exclude setup motions from results.
+        self.started = False
+
+        #This holds a refrence to the calling class
+        #self.theMotionBenchmark = None
+      
+        #The time at which the motion pause started
+        self.motionPauseStart = 0.0
+
+        #The time at which the current motion started
+        self.motionStartTime = 0.0
+
+        #The time at which the last iterative calculation was made
+        self.timeOfLastCalc = 0.0
+
+        #The number of test runs remaining
+        #self.runs = 0
+
+
+    #Just a distance calculation
+    def calcDist(self, point1, point2):
```

Comment:
function for distance between robocup points [here](https://github.com/RoboJackets/robocup-software/blob/staging/soccer/gameplay/robocup-py.cpp#L710)